monocle/github/pullrequest.py

```diff
@@ -257,6 +258,10 @@ def extract_pr_objects(pr):
                 change['repository_fullname'],
                 change['number'],
             )
+            change['url'] = ('https://%s/%s/pull/%s'
```

Comment:
I would have instead added three fields:
- host: could be github.com or github.company.com or gerrit.ovirt.com or softwarefactory-project.io
- host_type: could be gerrit or github
- host_base_url: should be None usually for most services accessible from '/' but must be '/r' for softwarefactory-project.io for instance.

From those fields, in the UI, we can compute the PR web url.

This three fields must be added in db.py self.index. A index deletion + re-index will be needed. And host_base_url it should be an option CLI argument in the crawler UI in bot gerrit and github crawler I guess.

Let me know if you think about a better way to achieve this.

Reply:
I think I'll prefer to have the url computed on the backend side to have the simplest API. Let's discuss when you'll have time.

Reply:
Ok fine to me, the change url could be computed in the in the crawler and be part of object in the DB. What about the url to access user profils ? Should it be computed in the same way ? I'd say yes if we move this way. We should also take care of the host_base_url.

Reply:
user profiles should be done the same way I'd say too. Should we just pass a base_url  to the crawlers?

Longer term we could also have internal pages for users and changes.

Reply:
Yes. And looking more at the code. I see we could use the existing --host option. For the gerrit crawler it is well handled, I can call the gerrit crawler this way:
```
python3 ./monocle/main.py gerrit_crawler --repository "^software-factory/.*" --updated-since "2019-01-01 00:00:00" --loop-delay 60 --host https://softwarefactory-project.io/r
```

But for github, it is not consistent because it does not include the scheme:
```
monocle github_crawler --host github.com --token XXX --org docker --updated-since 2020-03-01
```
So it should be a complete url with the scheme, and looking at graphql.py self.url is hardcoded. So this should be fixed as well.

So yes we could use --host directly.

Reply:
is it ok now?